exchanges/request/request.go

```diff
@@ -197,6 +197,12 @@ func (r *Requester) doRequest(req *http.Request, p *Item) error {
 		if err != nil {
 			return err
 		}
+		// Even in the case of an erroneous condition below, yield the parsed
+		// response to caller.
+		var unmarshallError error
+		if p.Result != nil && len(contents) > 0 {
```

Comment:
Wouldn't sending a non nil struct and then receiving no contents still be worthy of an error? Seems like the old way captures unexpected results